charts/aad-pod-identity/templates/mic-deployment.yaml

```diff
@@ -0,0 +1,123 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "aad-pod-identity.mic.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ template "aad-pod-identity.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ template "aad-pod-identity.chart" . }}
+    app.kubernetes.io/component: mic
+  annotations:
+    description: {{ .Chart.Description }}
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ template "aad-pod-identity.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/component: mic
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ template "aad-pod-identity.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
+        helm.sh/chart: {{ template "aad-pod-identity.chart" . }}
+        app.kubernetes.io/component: mic
+    spec:
+      {{- if .Values.rbac.enabled }}
+      serviceAccountName: {{ template "aad-pod-identity.mic.fullname" . }}
+      {{- end }}
+      containers:
+      - name: mic
+        image: "{{ .Values.image.repository }}/{{ .Values.mic.image }}:{{ .Values.mic.tag }}"
+        imagePullPolicy: {{ .Values.image.imagePullPolicy }}
+        args:
+          {{- if not .Values.adminsecret }}
+          - "--cloudconfig=/etc/kubernetes/azure.json"
+          {{- end }}
+          - "--logtostderr"
+          {{- if .Values.mic.logVerbosity }}
+          - -v={{ .Values.mic.logVerbosity }}
+          {{- end }}
+        env:
+          - name: FORCENAMESPACED
+            value: "{{ .Values.forceNameSpaced }}"
+          {{- if .Values.adminsecret }}
+          - name: CLOUD
+            valueFrom:
+              secretKeyRef:
+                key: Cloud
+                name: {{ template "aad-pod-identity.mic.fullname" . }}
+          - name: SUBSCRIPTION_ID
```

Comment:
does this mix the no azure.json and one with azure.json ?

Reply:
Yeah, it's all templated in mic deployment. If the admin secret exists, then we will populate all the env var for using no azure json. Otherwise we'll mount `/etc/kubernetes/azure.json` as usual.